lib/asciidoctor/path_resolver.rb

```diff
@@ -131,26 +131,46 @@ def initialize file_separator = nil, working_dir = nil
     @_partition_path_sys, @_partition_path_web = {}, {}
   end
 
-  # Public: Check if the specified path is an absolute root path.
-  # This operation considers posix paths, windows paths, and file URLs.
+  # Public: Check whether the specified path is an absolute path.
   #
-  # Unix absolute paths and UNC paths start with slash. Windows roots can start
-  # with a drive letter. Absolute paths may start with file://, http://, or
-  # https:// when the IO module is xmlhttprequest (Opal runtime only).
+  # This operation considers both posix paths and Windows paths. It does not
+  # consider URIs.
+  #
+  # Unix absolute paths and UNC paths both start with slash. Windows roots can
+  # start with a drive letter.
   #
   # path - the String path to check
   #
   # returns a Boolean indicating whether the path is an absolute root path
+  def absolute_path? path
+    (path.start_with? SLASH) || (@file_separator == BACKSLASH && (WindowsRootRx.match? path))
+  end
+
   if RUBY_ENGINE == 'opal'
-    def root? path
-      (path.start_with? SLASH) ||
-          (@file_separator == BACKSLASH && (WindowsRootRx.match? path)) ||
-          (::JAVASCRIPT_IO_MODULE == 'xmlhttprequest' && (path.start_with? 'file://', 'http://', 'https://'))
+    if ::JAVASCRIPT_IO_MODULE == 'xmlhttprequest'
```

Comment:
We could use a single `if` right ? Is there a particular reason why you choose to use two `if` ?

```rb
if RUBY_ENGINE == 'opal' && ::JAVASCRIPT_IO_MODULE == 'xmlhttprequest'
  def root?
    # ...
  end
else
  alias root? absolute_path?
end
```

Reply:
Because the second constant is not defined in the Ruby environment and thus will fail.

Reply:
Nevermind, I see what you're saying. You're right.

Reply:
I used to have another condition in there, but then I removed it. So now we can collapse it.